Sources/Hypertext.swift

```diff
@@ -6,18 +6,23 @@
 //  Copyright © 2016 Sahand Nayebaziz. All rights reserved.
 //
 
-public protocol Renderable {
-    func render() -> String
-    func render(startingWithSpacesCount: Int) -> String
```

Comment:
So I'm hearing you say a few things:

1. You'd like the simple `render()` method to be part of the protocol, because:
    1. You'd like the autocomplete to be completely clean and only show the two render functions
    * If it's part of the protocol (and not (just) the extension) people can overwrite it and provide their own implementation.
    * If there's a dedicated method for minified rendering, then the `indent` argument no longer needs to be nullable
* You'd like to swap the order of the spaces / indent, because it makes more sense to you that way.

Did I get you right? I totally agree with 1.i and 1.ii and I have some objections towards the rest of your philosophy.

I really value ease / simplicity and clarity at the call site. Like you said, I imagine that there are two main use cases for using this function:

1. On a website, where minified rendering is always preferred, this will be most common.
2. During debugging, where indented rendering is always preferred.

When I look at those use cases I think nobody really cares about being able to manipulate how the indentation will work. They'll just want it to work. Indented rendering for debugging purposes doesn't need to have a certain style, it just needs to be readable.

So this is what I did in the changes that I just committed:

* I kept the `indent` argument as an optional, because that just makes the code of the default implementations so much cleaner.
* I did put the `render()` method back into the protocol so people *can* implement their own version if they really want to, but I also kept it in the extension so people don't *have* to implement their own version.
* I changed the render method in such a way so that `startingWithSpacesCount` has become obsolete. This way the autocomplete stays cleaner, because that argument had no value at the call site at all.

So to summarize: I really want to keep things simple and easy for the developers using our library. I want a developer that implements their own custom tag to not have to worry about implementing a bunch of (render) methods that they don't care so much about anyway. And I want a developer that calls the `render` methods to not be distracted by arguments that are meaningless in that context, like `startingWithSpacesCount`.

Would you like to review my changes and tell me what you think?